plugins/obs-ffmpeg/obs-ffmpeg-hls-mux.c

```diff
@@ -0,0 +1,332 @@
+#include "obs-ffmpeg-mux.h"
+
+#define do_log(level, format, ...)                      \
+	blog(level, "[ffmpeg hls muxer: '%s'] " format, \
+	     obs_output_get_name(stream->output), ##__VA_ARGS__)
+
+#define warn(format, ...) do_log(LOG_WARNING, format, ##__VA_ARGS__)
+#define info(format, ...) do_log(LOG_INFO, format, ##__VA_ARGS__)
+
+const char *ffmpeg_hls_mux_getname(void *type)
+{
+	UNUSED_PARAMETER(type);
+	return obs_module_text("FFmpegHlsMuxer");
+}
+
+int hls_stream_dropped_frames(void *data)
+{
+	struct ffmpeg_muxer *stream = data;
+	return stream->dropped_frames;
+}
+
+void ffmpeg_hls_mux_destroy(void *data)
+{
+	struct ffmpeg_muxer *stream = data;
+
+	if (stream) {
+		deactivate(stream, 0);
+
+		pthread_mutex_destroy(&stream->write_mutex);
+		os_sem_destroy(stream->write_sem);
+		os_event_destroy(stream->stop_event);
+
+		da_free(stream->mux_packets);
+		circlebuf_free(&stream->packets);
+
+		os_process_pipe_destroy(stream->pipe);
+		dstr_free(&stream->path);
+		dstr_free(&stream->printable_path);
+		dstr_free(&stream->stream_key);
+		dstr_free(&stream->muxer_settings);
+		bfree(data);
+	}
+}
+
+void *ffmpeg_hls_mux_create(obs_data_t *settings, obs_output_t *output)
+{
+	struct ffmpeg_muxer *stream = bzalloc(sizeof(*stream));
+	pthread_mutex_init_value(&stream->write_mutex);
```

Comment:
I'm wondering if this is needed.

Reply:
the pthread_mutex_init_value call? Yeah I agree it seems strange/redundant but I observed it in almost every file where mutexes are used and thought there must be a reason. It's defined in threading.h which might shine some light.

Reply:
![image](https://user-images.githubusercontent.com/60421398/90040436-e648f400-dc95-11ea-9d5f-bd2bf9d8e5dc.png)
It dereferences our mutex and stores a mutex set to PTHREAD_MUTEX_INITIALIZER in it.